### PR TITLE
Add drill down view

### PR DESCRIPTION
**Pull request by philippjfr**
Allows selecting individual visits on the line plot and toggling between coadd view and visit view.

ToDo:

- [x] Load visit data when visit is selected
- [x] Decide whether toggle should be per filter or across filters (do all visits appear in all filters?) 

Screen Shot 2020-05-11 at 2 51 44 PM

Closes https://github.com/Quansight/lsst_dashboard/issues/110

### CODE REVIEW

File: lsst_dashboard/gui.py [hunk 1]
```diff
@@ -596,7 +604,7 @@ def _assemble_query_expression(self, ignore_query_expr=False):
 
         return query_expr
 
-    def get_dataset_by_filter(self, filter_type, metrics):
+    def get_dataset_by_filter(self, filter_type, metrics, visits=None):
```
Comment (philippjfr) [inline lsst_dashboard/gui.py:607]: This method is now given a list of selected visits (I think we only allow one). Should add a condition like:

```python
if visits and self.coadd_toggle.value == 'Visit View':
    return load_visit_data(filter_type, metrics, visits[0])
```

**Comment by dharhas** [comment]
@philippjfr #170 has the code to get the individual visit dataset. It is a very small change.

Example usage:

```python
from lsst_dashboard.dataset import Dataset
d = Dataset('/project/dharhas/DM-21335-New2-KTK', coadd_version='forced')
d.connect()
df = d.get_visits_by_metric_filter(filt='HSC-I', metric='Gaussian-PSF_magDiff_mmag', visit=14188)  
```

**Comment by philippjfr** [comment]
Perfect thanks. If you've tested it just merge and I'll rebase this PR on top of it.

**Comment by dharhas** [comment]
done.

**Comment by philippjfr** [comment]
@dharhas @timothydmorton This is ready for testing, not it is fairly slow until I've merged a PR fixing the getitem issues you were seeing before into HoloViews.

**Comment by timothydmorton** [comment]
Great!  With that getitem fixed, maybe we're still OK leaving data as dask ddfs rather than computing; I'll check that out too when I have the chance.

**Comment by philippjfr** [comment]
This is ready to merge btw, and should probably be merged before the black formatting stuff goes in.

**Pull request merged**